### ISSUE TITLE
Co-Start can not start up amazonEc2 services

### ISSUE THREAD

**Issue by charlietritran** [compatibleone/accords-platform#30]
Hi, 

I followed instructions from the README file of the download.  and do co-start and see the following error:
# 

Starting accords amazonEc2 services
Rest Client Request : GET /-/ HTTP/1.1 
Rest Client Request : GET /publication/ HTTP/1.1 
Rest Client Request : GET /publication/a8fa01f8-60b0-4926-9780-e8e6644e752f HTTP/1.1 
Rest Client Request : GET /-/ HTTP/1.1 
Rest Client Request : GET /user/ HTTP/1.1 
Rest Client Request : GET /-/ HTTP/1.1 
Rest Client Request : GET /publication/ HTTP/1.1 
Rest Client Request : GET /publication/a8fa01f8-60b0-4926-9780-e8e6644e752f HTTP/1.1 
Rest Client Request : GET /-/ HTTP/1.1 
Rest Client Request : GET /user/ HTTP/1.1 
Traceback (most recent call last):
  File "/usr/local/bin/ec2procci//ec2procci.py", line 21, in <module>
    import pycompdev
ImportError: No module named pycompdev
# 

assuming that i do not execute this section: 
## PYACCORDSSDK CONFIGURATION

To configure the PyAccordsSDK path, run the following (as root):
# ./pyaccords_config.sh

I would like to know if anyone can point me where should i look at?  i see that folder pycompdev in one of a dir in accords-platform.

thank you,
charlie

**Comment by charlietritran** [comment]
# if I execute that cmd, ./pyaccords_config.sh, then i got errors:

root@ubuntu:/home/trancha/bin/accords-platform# sh pyaccords_config.sh
running install
running build
running build_py
running install_lib
running install_egg_info
Removing /usr/local/lib/python2.7/dist-packages/pypacksrc-1.0.egg-info
Writing /usr/local/lib/python2.7/dist-packages/pypacksrc-1.0.egg-info
/home/trancha/bin/accords-platform
running install
running build
running build_ext
building 'pycompmanager' extension
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../coxml/src/ -I/usr/include/python2.7 -c pycompmanager.c -o build/temp.linux-x86_64-2.7/pycompmanager.o
In file included from pycompmanager.c:20:0:
compmanager.h:17:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
pycompmanager.c: In function ‘pycompmanager_removeCategory’:
pycompmanager.c:51:7: warning: unused variable ‘flag’ [-Wunused-variable]
pycompmanager.c: In function ‘pycompmanager_generateComponent’:
pycompmanager.c:72:9: warning: unused variable ‘flaglist’ [-Wunused-variable]
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../coxml/src/ -I/usr/include/python2.7 -c compmanager.c -o build/temp.linux-x86_64-2.7/compmanager.o
In file included from compmanager.c:19:0:
compmanager.h:17:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
compmanager.c:24:5: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../coxml/src/ -I/usr/include/python2.7 -c list.c -o build/temp.linux-x86_64-2.7/list.o
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../coxml/src/ -I/usr/include/python2.7 -c utools.c -o build/temp.linux-x86_64-2.7/utools.o
utools.c:21:22: fatal error: allocate.h: No such file or directory
compilation terminated.
error: command 'gcc' failed with exit status 1
/home/trancha/bin/accords-platform
running install
running build
running build_ext
building 'pycompdev' extension
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../occi/src/ -I../../cocarrier/src/ -I../../cocci/src/ -I../../coes/src/ -I../../command/src/ -I../../copabr/src -I../../cords/src/ -I../../corest/src -I../../coxml/src/ -I../../pubocci/src/ -I../../pyaccords/pysrc/ -I/usr/include/python2.7 -c pycompdev.c -o build/temp.linux-x86_64-2.7/pycompdev.o
In file included from pycompdev.c:19:0:
component.c:20:22: fatal error: standard.h: No such file or directory
compilation terminated.
error: command 'gcc' failed with exit status 1

# /home/trancha/bin/accords-platform

please tell me waht should i do to resolve those missing file errors?  i am using Python 2.7.3.

thank you.
charlie

**Comment by h-medjahed** [comment]
the pyaccordsSDK bug has been corrected

On Thu, Jan 23, 2014 at 9:52 PM, charlietritran notifications@github.comwrote:

> if I execute that cmd, ./pyaccords_config.sh, then i got errors:
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/compatibleone/accords-platform/issues/30#issuecomment-33167139
> .

## 

Cordialement;

---

## h.medjahed@gmail.com

**Comment by charlietritran** [comment]
hi h-medjahed,

i think those errors are gone but some new ones, about missing files, popped out:  i could not issue this command successfully:

./pyaccords_config.sh

here is what i have done:
- get latest sources
- install autoconf automake autotools make gcc libtool libssh-dev uuid uuid-dev libglobus (there are two items not found: autotools, libglobus ) 
- run: ./autogen.sh
- run: ./configure
- run: make clean
- run: make 
- run: make install
- run: ./pyaccords_config.sh

and here is the error that stops the further process:
=============== logged info ====================
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../coxml/src/ -I/usr/include/python2.7 -c list.c -o build/temp.linux-x86_64-2.7/list.o
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../coxml/src/ -I/usr/include/python2.7 -c utools.c -o build/temp.linux-x86_64-2.7/utools.o
utools.c:21:22: fatal error: allocate.h: No such file or directory
compilation terminated.
error: command 'gcc' failed with exit status 1
/home/trancha/bin/accords-platform
running install
running build
running build_ext
building 'pycompdev' extension
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../occi/src/ -I../../cocarrier/src/ -I../../cocci/src/ -I../../coes/src/ -I../../command/src/ -I../../copabr/src -I../../cords/src/ -I../../corest/src -I../../coxml/src/ -I../../pubocci/src/ -I../../pyaccords/pysrc/ -I/usr/include/python2.7 -c pycompdev.c -o build/temp.linux-x86_64-2.7/pycompdev.o
In file included from pycompdev.c:19:0:
component.c:20:22: fatal error: standard.h: No such file or directory
compilation terminated.
error: command 'gcc' failed with exit status 1
/home/trancha/bin/accords-platform

# 

i can trace that error from the content of this file: pyaccords_config.sh, by following each step in the list:

cd pyaccords/pysrc/pypacksrc
python setup.py install 
cd -
cd pyaccordsSDK/pycompmanager
python setup.py install // produce this error: utools.c:21:22: fatal error: allocate.h: No such file or directory
cd -
cd pyaccordsSDK/pycompdev
python setup.py install // produce this error: component.c:20:22: fatal error: standard.h: No such file or directory
cd -

i would like to have more info on how to resolve these?  of courses, i can run the platform without those python setup stuffs and be able to provision instances on openstack.

**Comment by h-medjahed** [comment]
Hi Charlie
Errors are gone now, you have just to procceed correctly,
in order to issues this command
   # ./pyaccords_config.sh
before this you need to run
 #make
 #make install
because some PyaccordsSdk's dependency files and libraries are generated
during these steps (like compute.h for example ...)

Regards,
Hamid Medjahed

On Fri, Jan 24, 2014 at 7:32 PM, charlietritran notifications@github.comwrote:

> hi h-medjahed,
> 
> i think those errors are gone but there is one new error, about missing
> file, popped out: i could not issue this command successfully:
> 
> ./pyaccords_config.sh
> and here is the error that stops the further process:
> 
> ../../cocci/occiclient.h:104:15: warning: function declaration isn’t a
> prototype [-Wstrict-prototypes]
> ../../cocci/occiclient.h:118:16: warning: function declaration isn’t a
> prototype [-Wstrict-prototypes]
> ../../cocci/occiclient.h:120:1: warning: function declaration isn’t a
> prototype [-Wstrict-prototypes]
> In file included from ../../occi/occibuilder.h:5:0,
> from component.c:26,
> from pycompdev.c:19:
> ../../cords/cords.h:21:21: fatal error: compute.h: No such file or
> directory
> compilation terminated.
> error: command 'gcc' failed with exit status 1
> 
> i can trace that error from the content of this file: pyaccords_config.sh,
> by following each step in the list:
> 
> cd pyaccords/pysrc/pypacksrc
> python setup.py install
> cd -
> cd pyaccordsSDK/pycompmanager
> python setup.py install
> cd -
> cd pyaccordsSDK/pycompdev
> python setup.py install
> cd -
> and see that error is from the last "install":
> 
> cd pyaccordsSDK/pycompdev
> python setup.py install
> 
> so, can you give me more info on how to resolve it?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/compatibleone/accords-platform/issues/30#issuecomment-33248337
> .

## 

Cordialement;

---

## h.medjahed@gmail.com

**Comment by charlietritran** [comment]
Hi Hamid,

I am sorry I could not make it works as you said.  I try a new VM, Ubuntu Server 12.04, and new accords-platform downloaded.  after install all needed libs, i starts from root dir of the platform:

1) ./autogen.sh
2) ./configure
3) ./ make
4) ./ make install
5) ././pyaccords_config.sh

then, still see the same missing:
 allocate.h, and standard.h.

Please tell me what am i missing?  

Thanks in advance,
charlie

**Comment by ijm667** [comment]
Hello Charlie,
I just went through the same procedure on a new debian box, ( i dont have a ubuntu ) and it all worked fine.
Do you have python 2.7 installed ?
SincerelyJamie

Date: Sat, 25 Jan 2014 13:50:35 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

I am happy to hear that it is working on your debian box.  That means amazonEc2 srv is working fine, regarding to starting up accords-platform's services.  So, the issue may be with the OS, this Ubuntu server 12.04 LTS.  Yes, I have python, but 2.7.3, not 2.7, and I don't really know for sure what is the difference between the twos.  I will keep tracing to see why those two files are missing as well as where they supposed to be located.  

Thank you for your time,
Charlie

**Comment by charlietritran** [comment]
Hi Jamie,

I just want to make sure that are those two files, allocate.h and standard.h in the folder, coxml?  if so, i can see them, but not sure why this cmd, ./pyaccords_config.sh, can not see them.

thanks,
charlie

**Comment by ijm667** [comment]
Hello Charlie,
I have been looking at your problem here and I now see something that is not right in the EC2 PROCCI Generation on my debian platform. For some reason it was not being picked up before.
First off, what is the production model ? Are you using MySql or PostGre SQL Database as back end or are you in the XML Model ?
I have had to modify the setup.py (  .../accords-platform/pyaccordsSDK/pycompdev/setup.py ) to add the path../../cosql to the list of include paths. This has solved the compile error for me.Check this out and let me know please.
SincerelyJamie

Date: Tue, 28 Jan 2014 07:41:40 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

I followed your direction by adding that dir to the setting.py and it solved the issue with missing files.  I think i am using this dir, coxml.  i do not see cosql like yours.  but now, a new missing file popped out, rest.h.  so, i have to look into this similar problem again.  

Thank you for the hint,
Charlie

**Comment by ijm667** [comment]
Hello Charlie,
I am intrigued .... would it be possible to have a remote login to your machine to see what is going on ?
SincerelyJamie

Date: Tue, 28 Jan 2014 10:19:46 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

I am almost gave up for troubleshoot the ec2 service's issue.  it is still not started up yet.  my laptop is not allowed to ssh from outside.  so, can we do some screen-leaf sometimes later next week?  or you can provide me some hints as where to look at?
I would like to make sure if we are in the same scenarios, regarding to the setup of accords-platform.  Here is what i have done:
- start a virtual machine, with ubuntu 12.04 OS installed, with git, and openssh-server.
- from my_user folder, make bin dir, and move to bin. (my_user is the USER folder for current user)
- using git to clone the flatform to my_user folder.
- installed autoconf, automake, make, libssl-dev, uuid, uuid-dev, python-dev, python-boto
- do: ./autogen.sh => working fine
- do: ./configure => working fine
- do: make
- do: sudo make install
- do: sudo ./pyaccords_config.sh
  then i see errors for two missing files, allocate.h and standard.h.  those two files are both in the accords-platform/coxml.

## \- then i modified this file, accords-platform/pyaccordsSDK/pycompmanager/setup.py, to add ../../coxml to the include dir, and that only help for the allocate.h missing.

module1 = Extension('pycompmanager', 
                     sources = ['pycompmanager.c','compmanager.c','list.c','utools.c'],
                     depends = ['compmanager.h','list.c'],
                     include_dirs = ['../../coxml/src/', '../../coxml'],
             libraries = ['coxml'],
                     library_dirs = ['../../coxml/src/.libs/', '../../coxml/.libs/']

##                    )
- redo: sudo ./pyaccords_config.sh and still have the standard.h missing:

trancha@ubuntu:~/bin/accords-platform$ sudo ./pyaccords_config.sh
running install
running build
running build_py
running install_lib
running install_egg_info
Removing /usr/local/lib/python2.7/dist-packages/pypacksrc-1.0.egg-info
Writing /usr/local/lib/python2.7/dist-packages/pypacksrc-1.0.egg-info
/home/trancha/bin/accords-platform
running install
running build
running build_ext
running install_lib
running install_egg_info
Removing /usr/local/lib/python2.7/dist-packages/pycompmanager-1.0.egg-info
Writing /usr/local/lib/python2.7/dist-packages/pycompmanager-1.0.egg-info
/home/trancha/bin/accords-platform
Traceback (most recent call last):
  File "setup.py", line 22, in <module>
    ext_modules = [module1])
NameError: name 'module1' is not defined
/home/trancha/bin/accords-platform
trancha@ubuntu:~/bin/accords-platform$ sudo ./pyaccords_config.sh
running install
running build
running build_py
running install_lib
running install_egg_info
Removing /usr/local/lib/python2.7/dist-packages/pypacksrc-1.0.egg-info
Writing /usr/local/lib/python2.7/dist-packages/pypacksrc-1.0.egg-info
/home/trancha/bin/accords-platform
running install
running build
running build_ext
running install_lib
running install_egg_info
Removing /usr/local/lib/python2.7/dist-packages/pycompmanager-1.0.egg-info
Writing /usr/local/lib/python2.7/dist-packages/pycompmanager-1.0.egg-info
/home/trancha/bin/accords-platform
running install
running build
running build_ext
building 'pycompdev' extension
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I../../occi/src/ -I../../cocarrier/src/ -I../../cocci/src/ -I../../coes/src/ -I../../command/src/ -I../../copabr/src -I../../cords/src/ -I../../corest/src -I../../coxml/src/ -I../../pubocci/src/ -I../../pyaccords/pysrc/ -I/usr/include/python2.7 -c pycompdev.c -o build/temp.linux-x86_64-2.7/pycompdev.o
In file included from pycompdev.c:19:0:
component.c:20:22: fatal error: standard.h: No such file or directory
- and now, if i add the same thing, ../../coxml to the setting.py in accords-platform/pyaccordsSDK/pycompdev, then that standard.h missing goes away, but a new one popped out, rest.h.  
- if i keep adding the folder (rest.h is in corest folder) of the missing file to the setup.py, then a new one xxx.h will be missing while the old missing is resolved.

so, I think i may make something wrong, but i do not see what i missed.

Can you have any advice?  I even tried with debian version 7, and got the same result.

What I have found so far for accords-platform is it is still working without the pyaccords_config, and i can provision instance for openstack.  But for amazon ec2 provision, i think i need that config to start up the service.  and now, i got stuck.

Sincerely,
Charlie

**Comment by charlietritran** [comment]
Hi Jamie,

I think it is working now for that cmd, ./pyaccords_config.sh.  But I had to make many changes / adjust for the include paths/files in the setup.py and a bunch of other .h/.c files in folder pyaccordsXXX.  I am not sure if my changes will affect to the funcitonalities of the platform, regarding to the provision features.  I am testing it now and see the ec2procci service is up and ready after co-start.

thank you,
charlie

**Comment by ijm667** [comment]
Hello Charlie,
If you want me to check out the changes that you had to make please forward them to me by email and I will have a look. I am sorry to hear that you have had such trouble with the EC2 PROCCI.I know that the version level of the python library caused me a lot of initial trouble but once through that it all worked fine.I have actually stress tested that component particularily and it has been working just fine.
SincerelyJamie

Date: Thu, 30 Jan 2014 07:45:36 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

I have not completely tested the ec2procci service because some issue with my own aws's acct, not from accords-platform.  I will try again after my acct is resolved.  in the mean time, you can review what I have changed that make the ./pyaccords_config.sh run successful.  assuning I  am using xml model (because I  do not see cosql in the platform dir after done git clone ... to my vm as opposite to the download directly for master version), here is the changes:
- update include_dirs for setup.py in accords-platform/pyaccordsSDK/pycompmanager with:
  include_dirs = ['../../coxml/src/', '../../coxml'],
- update include_dirs for setup.py in accords-platform/pyaccordsSDK/pycompdev with:
  include_dirs=['../../occi/src/','../../cocarrier/src/','../../cocci/src/','../../coes/src/','../../command/src/','../../copabr/src','../../cords/src/','../../corest/src','../../coxml/src/','../../pubocci/src/','../../pyaccords/pysrc/','../../coxml', '../../corest', '../../occi', '../../cords','../../pubocci', '../../cocci', '../../copabr'],
- replace #include "../../occi/src/occi.h" with #include "../../occi/occi.h" for any .h / .c file in this dir: accords-platform/pyaccords/pysrc that applicable

In summary, there is no code changed, but references to file path / include dirs that needed to be updated in some setup.py and .h / .c files.  May be this is due to python version.  mine is 2.7.3.  I am using the source code and do compilation.  OS is ubuntu 12.04 LTS

after those changes, the platform can start up ec2procci srv successfully, and this, I believe, will take care of providing REST's service calls to aws, regarding to the provisioning task.

Please advice me if I am on the wrong way.

thank you,
charlie

**Comment by ijm667** [comment]
Hello Charlie,
I just read through this and am a little surprised that you don't have the cosql directory. This is really quite important because it represents a huge step forward in the industrialisation of the Accords Platform and all major work is being performed on this version since. I would suggest working from the master as quickly as possible to avoid you being out on a limb so to speak.
My version of Python here is the 2.7.3 so that shouldn'tbe an issue for you.
As a guide:
I create a directory ( somewhere for example /home/c2 )
I change into that directory 
I clone the sources from git
I set up the CFLAGS="-g -ggdb -o0" to ensure ease of debug
I launch the following commands from inside the /home/c2/accords-platform directory
./autogen.sh./configuremakemake install
I then step back from the accords-platform directory into the /home/c2 directory and launch
accords-platform/scripts/co-install small-accords
this creates and starts a working version of the platform in the small-accords platform directory
How does this work for you ?
I have just been through this procedure on four different clean machines and it is working for me.
SincerelyJamie

Date: Fri, 31 Jan 2014 06:57:36 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

I have to download the .zip file, accords-platform-master, so as to be able to have the dir cosql.  Now, I got a complain / error for ./configure run:

checking ldap.h usability... no
checking ldap.h presence... no
checking for ldap.h... no
configure: error: You are missing ldap development files

can you advice me on how to fix this?  I do not know why the git clone does not pull all needed files for me.

thanks,
charlie

**Comment by ijm667** [comment]
Hi Charlie,
what kind of failure ( postgre sql ? mysql ? ) you will need to modify the configure.ac to remove either the mysql stuff or the postgre sql stuff and install the required system packages to handle the one of your choice. I would suggest using mysql as it has been the most tested.
sincerelyJamie 

Date: Fri, 31 Jan 2014 10:36:08 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

I am not clear as to what need to be changed in the configure.ac?  I can use either mysql or postgres.  but as you recommended, I can use mysql.  But I do not know what should I do in that .ac file?  is that the reason for this missing error:

configure: error: You are missing ldap development files

thanks,
charlie

**Comment by ijm667** [comment]
Hello Charlie,
yes this would be cause of the problem. The configure.ac is delivered set up to use either MySql or PostGreSql as the backend database for the storage of category instance information. In previous versions this was done using an XML flat file management system. The use of the new SQL system is important for many reasons so you will need to choose which one to use. I recomend the MYSQL because the it has been most tested until now.

Keep either this for MYSQL 
AC_CHECK_HEADERS([mysql/mysql.h],                [break],                [AC_MSG_ERROR([You are missing mysql development files])])
Or this for POSTGRE SQL
AC_CHECK_HEADERS([postgresql/libpq-fe.h],                [break],                [AC_MSG_ERROR([You are missing postgresql development files])])
Then .....
Keep either this for MYSQL
if test "$(uname -s)" = 'Linux' ; then        AC_SUBST([LDFLAGS_MYSQL], ["-lmysqlclient_r"])fi
Or this for POSTGRE SQL
if test "$(uname -s)" = 'Linux' ; then        AC_SUBST([LDFLAGS_POSTGRE], ["/usr/lib/libpq.so"])fi

You can either comment out or delete the corresponding section of the configure.ac file.
Then ....
Edit the scripts/accords.in file to ensure that the following correspond to your needs
resthost=127.0.0.1export OCCISQLTYPE=MYSQLexport OCCISQLHOST=127.0.0.1export OCCISQLUSER=rootexport OCCISQLPASSWD=root
Is that clearer ? 
You will also need to install the corresponding Database Packages depending on your system version.
SincerelyJamie

Date: Fri, 31 Jan 2014 10:50:04 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

I tried to mod those two files, and redo: ./aufogen.sh, ./configure and still got that ldap missing error.   I will try to find out why.  

---

trancha@cloud-broker:~/bin/accords-platform$ ./configure
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking how to create a pax tar archive... gnutar
checking whether make supports nested variables... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking whether make sets $(MAKE)... (cached) yes
checking whether ln -s works... yes
checking for ANSI C header files... (cached) yes
checking for SSL library... /usr
checking ldap.h usability... no
checking ldap.h presence... no
checking for ldap.h... no
configure: error: You are missing ldap development files

---

Thanks,
Charlie

**Comment by ijm667** [comment]
Charlie,
one of the following should rectify that problem
aptitude install libldap
aptitude install libldap2aptitude install libldap2-dev
or something similar that can be found using
aptitude search ldap
fingers corssed for you
SincerelyJamie

Date: Fri, 31 Jan 2014 11:31:13 -0800

**Comment by charlietritran** [comment]
Hi Jamie,

The master downloaded is working now.  I can see a new folder small-accords be generated.  here is what i have done, based on your instructions:
- download the zip from this site, 
- expand the accords-platform-master to a local dir, say HOME/myfolder
- move to HOME/myfolder/accords-plaform-master 

## \- edit configure.ac by removing the postgres db sections as indicated in your posted:

## Keep either this for MYSQL AC_CHECK_HEADERS([mysql/mysql.h], [break], [AC_MSG_ERROR([You are missing mysql development files])]) Or this for POSTGRE SQL AC_CHECK_HEADERS([postgresql/libpq-fe.h], [break], [AC_MSG_ERROR([You are missing postgresql development files])]) Then ..... Keep either this for MYSQL if test "$(uname -s)" = 'Linux' ; then AC_SUBST([LDFLAGS_MYSQL], ["-lmysqlclient_r"])fi Or this for POSTGRE SQL if test "$(uname -s)" = 'Linux' ; then AC_SUBST([LDFLAGS_POSTGRE], ["/usr/lib/libpq.so"])fi You can either comment out or delete the corresponding section of the configure.ac file. Then .... Edit the scripts/accords.in file to ensure that the following correspond to your needs resthost=127.0.0.1export OCCISQLTYPE=MYSQLexport OCCISQLHOST=127.0.0.1export OCCISQLUSER=rootexport OCCISQLPASSWD=root
- assuming to use mysql, then install libldap2-dev, libmysqld-dev
- now, from platform dir, run ./autogen.sh,  ./configure, make clean, make, and sudo make install and sudo ./pyaccords_config.sh
- at this time, everything should be fine, no error.
- move back to HOME/mydir and do sudo accords-platform-master/scripts/co-install small-accords
- there will be a new dir small-accords be generated under HOME/mydir after the install.

that is all.

thank you,
charlie

**Comment by ijm667** [comment]
Hello Charlie,
that is great news. I am glad that you have got it working on your laptop now and thank you for the precious feedback.
SincerelyJamie
Date: Fri, 31 Jan 2014 20:57:01 -0800

**Comment by ghost** [comment]
Hello,

We have tried to install the accord platform, and followed the steps indicated in this post. However when running the command "make" we get this error:

Making install in cosched
make[1]: Entering directory /home/accords-platform/cosched' CCLD cosched /home/accords-platform/cocspi/.libs/libcocspi.so: undefined reference toselect_occi_sql_records'
collect2: ld returned 1 exit status
make[1]: **\* [cosched] Error 1
make[1]: Leaving directory `/home/accords-platform/cosched'
make: **\* [install-recursive] Error 1

Thank you for your help

**Comment by charlietritran** [comment]
hi,

I do not use Comp-One any more after we are able to send REST srvs to Amazon Ec2 via that broker and personally have not seen those errors other than what I have seen and already put in the post.
- did you have the same os as mine, Ubuntu 12.-4 LTS?  I think CompatibleOne is developed on Centos6.
- did you download and use the master from git-repos?
- did you do config as instructed?

you may need to learn python if not already known, so as to look at the files if needed for tracing the errors and be able to fix them by your own.

from this error, undefined reference toselect_occi_sql_records', i guess it is related to db.  you need to know what kinds of db your comp-one is using.

what I have done is to make a copy of a vm after install os and keep the original one.  then, I will install and config Comp-One on the copy and see if it is not working or messed up, I will delete that cp of vm and make another copy to make things to be clean.

good luck,
Charlie

Date: Tue, 29 Apr 2014 05:17:18 -0700

**Comment by ghost** [comment]
Hey,

Thanks for the reply, I deleted the previous VM and created a new clean one that run a debian OS, and retried with the same steps and now it seems to work, and I have the accords-platform working now. I followed the instructions mentioned in this post and I used the accords-platform-master from git-repos

But now I want to create a new provider (I have another VM running OpenStack), parse a manifest and finally deploy the VM described in this manifest. In order to do that I use the account "accords" already available in accounts.xml in the folder "small accords" and then:
- run: co-stop
- configure: os_config.xml
- run co-start

Everything seems to be running fine, I can also communicate with the other VM running OpenStack using the command "testos".

But whenever I try to parse the manifest with the command "co-parser" I get the following error:

Rest Client Request : GET /-/ HTTP/1.1 
Rest Client Request : GET /publication/ HTTP/1.1 
Rest Client Request : GET /publication/c47eece0-00b4-4763-b502-394429296500 HTTP/1.1 
Rest Client Request : GET /publication/5d43f2c8-d030-43fb-8fc6-c74258a6123e HTTP/1.1 
Rest Client Request : GET /-/ HTTP/1.1 
Rest Client Request : GET /user/ HTTP/1.1 
Rest Client Request : GET /user/b949011b-pars-400c-b6a7-c3f8a17f0ee6 HTTP/1.1 
Rest Client Request : GET /publication/ HTTP/1.1 
Rest Client Request : GET /publication/f90ad453-0e2c-4bd7-b724-39b973b8e53c HTTP/1.1 
Rest Client Request : GET /publication/954d42b2-26f5-4318-8147-0b8bfe15fa3a HTTP/1.1 
Rest Client Request : POST /authorization/ HTTP/1.1 
   CORDS Request Parser Phase 1

cords: error(4): parse error mymanifest.xml
Rest Client Request : DELETE /authorization/55a229ff-6d8b-4bc1-a275-f451e92f4e4e HTTP/1.1

My os_config.xml file:

<os_configs>
<os_config
 id="75a5a0f0-aa01-43b1-9873-e9c9c63bd142" orderid="" tls="" authenticate="" name="accords" namespace="admin" agent="OSPROCCI/1.0a" current="0" host="http://10.0.2.18:5000/v2.0" version="v1.1" user="admin" base="" password="admin" description="OpenStack" />
</os_configs>

My manifest:

<?xml version="1.0" encoding="UTF8"?>
<manifest name="mymanifest" xmlns="http://www.compatibleone.fr/schemes/manifest.xsd">
        <node name="node_test" type="simple" access="public" scope="normal" provider="any">
                <infrastructure name="os">
                        <compute name=":small" architecture="x86_64" cores="1" memory="1G" speed="1G"/>
                        <storage name=":small" size="10G"/>
                        <network name="compatibleone" label="ethernet" vlan="100M">
                                <port name="ssh" protocol="tcp" from="22" to="22"/>
                                <port name="cosacs" protocol="tcp" from="8286" to="8286"/>
                                <port name="http" protocol="tcp" from="80" to="80"/>
                        </network>
                </infrastructure>
                <image name=":cirros" agent="none">
                        <system name="cirros" />
        </image>
        </node>
        <configuration/>
        <release/>
        <interface/>
        <account name="accords"/>
        <security/>
</manifest>

Thanks for the help.
Elie.

**Comment by ghost** [comment]
It appears there is a problem showing the manifest in the post so I attached it as an image In this post. Sorry for this inconvenience.
![screenshot 2014-04-30 16 47 14](https://cloud.githubusercontent.com/assets/7438603/2842617/7fa1e824-d076-11e3-8599-20ad5db836ea.png)
